<launch>

    <!-- Arguments -->
    <arg name="map_file" default="/home/khiem/map/robotics_lab.yaml"/>
    <arg name="open_rviz" default="true"/>

    <!-- Robots' names -->
    <arg name="turtle_1" default="turtlebot01"/>
    <arg name="turtle_2" default="turtlebot02"/>
    <arg name="turtle_3" default="turtlebot03"/>

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

    <!-- Navigator -->
    <node pkg="multi_turtlebot3_navigation" name="navigator" type="navigator"/>

    <!-- Launch 2 robots -->
    <group ns="$(arg turtle_1)"> 
        <include file="$(find multi_turtlebot3_navigation)/launch/single_robot.launch">
            <arg name="multi_robot_name"  value="$(arg turtle_1)"/>
            <arg name="initial_pose_x"    value="0.0"/>
            <arg name="initial_pose_y"    value="0.0"/>
            <arg name="initial_pose_a"    value="0.0"/>
            <arg name="remap_goal"        value="false"/>
            <arg name="remap_initialpose" value="false"/>
        </include>
    </group>
    <group ns="$(arg turtle_2)"> 
        <include file="$(find multi_turtlebot3_navigation)/launch/single_robot.launch">
            <arg name="multi_robot_name"  value="$(arg turtle_2)"/>
            <arg name="initial_pose_x"    value="1.0"/>
            <arg name="initial_pose_y"    value="1.0"/>
            <arg name="initial_pose_a"    value="0.0"/>
            <arg name="remap_goal"        value="false"/>
            <arg name="remap_initialpose" value="false"/>
        </include>
    </group>
    <group ns="$(arg turtle_3)"> 
        <include file="$(find multi_turtlebot3_navigation)/launch/single_robot.launch">
            <arg name="multi_robot_name"  value="$(arg turtle_3)"/>
            <arg name="initial_pose_x"    value="2.0"/>
            <arg name="initial_pose_y"    value="2.0"/>
            <arg name="initial_pose_a"    value="0.0"/>
            <arg name="remap_goal"        value="false"/>
            <arg name="remap_initialpose" value="false"/>
        </include>
    </group>

    <!-- rviz -->
    <group if="$(arg open_rviz)"> 
        <node pkg="rviz" type="rviz" name="rviz_1" required="true"
            args="-d $(find multi_turtlebot3_navigation)/rviz/multi_turtlebot3_navigation.rviz"/>
    </group>

</launch>
